File: Exercises_Python/1961.py


altura_pulo, qnt_canos = map(int, input().split())

canos = list(map(int, input().split()))


sapo_pula = True
for i in range(0,qnt_canos - 1):
    if canos[i+1] - canos[i] > altura_pulo:
        sapo_pula = False
    elif abs(canos[i] - canos[i+1]) > altura_pulo:
        sapo_pula = False

if sapo_pula:
    print('YOU WIN')
else:
    print('GAME OVER')

File: Exercises_C++/1961.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

	int pulo, tamanho,numero;
	cin >> pulo >> tamanho;

	int vetor[tamanho];

	for (int c=0 ; c < tamanho ; c++) {
		cin >> numero;
		vetor[c] = numero;
	}
	
	bool winner;
	winner = true;
	for (int k=0 ; k < (tamanho-1) ; k++) {
		if (abs(vetor[k+1] - vetor[k]) > pulo){
			winner = false;
			break;
		}
	}
	
	if (winner) {
		cout << "YOU WIN" << endl;
	}
	else {
		cout << "GAME OVER" << endl;
	}

}
